endpoint.cpp

```diff
@@ -344,6 +344,48 @@ void Endpoint::print_statistics()
     printf("\n}\n");
 }
 
+void Endpoint::untrim_msg(const struct buffer *from, struct buffer *to)
```

Comment:
Since this is for our internal use... why don't we just mark that the message is trimmed rather than having to copy it around?